src/jackdaw/test/fixtures.clj

```diff
@@ -41,19 +45,83 @@
     (-> (.createTopics client required)
         (.all))))
 
+(def default-topic-opts {:timeout-ms 10000
+                         :delete-first? false})
+(defn parse-opts
+  [opts]
+  (if (integer? opts)
+    (assoc default-topic-opts :timeout-ms opts)
+    (merge default-topic-opts opts)))
+
 (defn topic-fixture
   "Returns a fixture function that creates all the topics named in the supplied
    topic config before running a test function."
   ([kafka-config topic-config]
-   (topic-fixture kafka-config topic-config 10000))
+   (topic-fixture kafka-config topic-config {}))
 
-  ([kafka-config topic-config timeout-ms]
+  ([kafka-config topic-config opts]
    (fn [t]
-     (with-open [client (AdminClient/create kafka-config)]
-       (-> (create-topics client kafka-config topic-config)
-           (.get timeout-ms java.util.concurrent.TimeUnit/MILLISECONDS))
-       (log/info "topic-fixture: created topics: " (keys topic-config))
-       (t)))))
+     (let [{:keys [timeout-ms delete-first?]} (parse-opts opts)]
```

Comment:
`parse-opts` is introduced in order to maintain compatibility with the existing version. Callers that provide an integer for timeout-ms will continue to behave the same way.